map_server/src/image_loader.cpp

```diff
@@ -99,7 +99,7 @@ loadMapFromFile(nav_msgs::GetMap::Response* resp,
   rowstride = img->pitch;
   n_channels = img->format->BytesPerPixel;
 
-  if (trinary || n_channels == 1)
+  if (n_channels <= 3)
```

Comment:
I think this should actually be "mode==Trinary || n_channels == 1" -- I'm not even sure why mode==Trinary is there, but let's not change the functionality of trinary mode